client/lib/media/utils.js

```diff
@@ -111,7 +112,7 @@ export function getFileExtension( media ) {
 	if ( isString ) {
 		let filePath;
 		if ( isUri( media ) ) {
-			filePath = urlLib.parse( media ).pathname;
+			filePath = new URL( media, BASE_PARSE_URL ).pathname;
```

Comment:
here's that risk of throwing an uncaught exception

in fact, I wonder here if this change warrants more refactoring, because it looks like `isUri` would be obviated by a robust URL parser

```js
if ( isString ) {
	const filePathUrl = parseURL( media, BASE_PARSE_URL );
	const filePath = filePathUrl ? filePathUrl.pathname : media;
	
	extension = path.extname( filepath ).slice( 1 );
}
```

Reply:
There is no risk here, because we already used `isUri` to determine if the string is a valid URL. Since we're using a base path, this won't throw on partial URLs, and this won't fail at all.

`isUri` provides different checks which are needed in this case. In particular, it looks for invalid characters in the URL string which would be otherwise ignored. Since this method is trying to handle both URLs and filesystem locations, we need the stricter validation up top, and can't just rely on `URL`'s validations.